src/Mvc/Mvc.Formatters.Json/src/Internal/JsonResultExecutor.cs

```diff
@@ -128,9 +128,11 @@ public virtual Task ExecuteAsync(ActionContext context, JsonResult result)
                     var jsonSerializer = JsonSerializer.Create(serializerSettings);
                     jsonSerializer.Serialize(jsonWriter, result.Value);
                 }
-            }
 
-            return Task.CompletedTask;
+                // Perf: call FlushAsync to call WriteAsync on the stream with any content left in the TextWriter's
+                // buffers. This is better than just letting dispose handle it (which would result in a synchronous write).
+                await writer.FlushAsync();
```

Comment:
`await using(var writer = ...` instead to call `DisposeAsync`?

Reply:
Cherry-picked the commit. Plus this is 2.1, I think it uses C# 7 \ `netcoreapp2.1` which likely doesn't support using either?